### ISSUE TITLE
AccordionBox: does hiding the title when expanded present a11y issues?

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/sun#339]
AccordionBox has a feature where the title can be hidden when the box is expanded.  Some sims use this feature when they are tight on space.  Some use it as an aesthetic choice.   At 3/1/18 dev meeting, there was discussion about whether hiding the title presents a11y problems for users.  @jessegreenberg was going to followup with @terracoda.

I'm interested in this because Equality Explorer (new sim, in development) hides titles for 2 of its 3 AccordionBoxes.  And the decision to hide title in this case falls into the "aesthetic choice" category.  See for example the "Equation or Inequality" and "Variable" accordion boxes in the "Variables" screen of [1.0.0-dev.67](https://www.colorado.edu/physics/phet/dev/html/equality-explorer/1.0.0-dev.67/phet/equality-explorer_en_phet.html). @amanda-phet FYI.

**Comment by jessegreenberg** [comment]
I will bring this up for discussion at the next a11y meeting, which is 10am 3/16/18.

**Comment by jessegreenberg** [comment]
We discussed this on 3/16/18. @terracoda said that as long as the close button touch area is plenty big and the button is well labeled for screen reader users, it is not an issue for a11y to make the title invisible when the AccordionBox is open.

Back to @pixelzoom to review, but no changes for a11y are recommended.

**Comment by pixelzoom** [comment]
Thanks for following up @jessegreenberg.

Labeling for dev meeting to share with other devs.

**Comment by terracoda** [comment]
@jessegreenberg and @pixelzoom, one more thought on this interaction. You both may have already thought about this, but just in case, I will mention it here. 

When opening and closing an accordion box, there is no change in context, only a change in content, so focus should remain on the triggering element, the heading.

In the case where the heading text is visually remove, we will likely need to ensure that it is visually hidden but not gone. We will still need the content there for screen readers users, even if the text is hidden.

Does that make sense. I can explain further if need me.

**Comment by jessegreenberg** [comment]
Understood @terracoda, we can make sure focus is managed in this way and make sure that the content is there for screen readers even if invisible.

**Comment by terracoda** [comment]
All that said, making the heading disappear does put a higher cognitive load on visual users. We may need to investigate that aspect further. I was thinking about non-visual access.

**Comment by amanda-phet** [comment]
>All that said, making the heading disappear does put a higher cognitive load on visual users. 

@jessegreenberg do you mean because the user needs to "remember" what the title was? I'll just throw out there that sometimes we put content in a panel and later decide we'd like to minimize that panel. When we turn it into an accordion box we are forced to come up with a title, and if not for that requirement we would never have included a title.

**Comment by terracoda** [comment]
Thanks for chiming in @amanda-phet, I was going to ask who is the deisgner in this sim ;-)
Yes, I did mean, forcing a user to remember causes cognitive load. 

If you think the title is almost unnecessary (i.e., content of open accordion is self-explanatory) than there should be no cognitive load on the user when the box is expanded and the title is hidden.

**Comment by amanda-phet** [comment]
I think we are talking generally about sims, but the original post was about Equality Explorer and yes I'm the lead on that sim :) 

Another example you should look at is Area Builder (https://phet.colorado.edu/sims/html/area-builder/latest/area-builder_en.html). In this sim, the title (Values) doesn't offer anything useful about the content (Area= and Perimeter= ). It was clearly put in there so that there was a title when minimized, but is very unnecessary when maximized. 

This issue came up with Area Model Introduction, though, and in that sim we decided to always show the Calculation title. I think that was the right call, because in that case the title might be useful for a user, and is not problematic for users who don't find the title useful.

**Comment by pixelzoom** [comment]
Discussed this a bit at 3/29/18 developer meeting. To summarize:

• Hiding the title when expanded doesn't cause a11y problems.
• When expanding/collapsing the accordion box, a11y focus should remain on the triggering element (the title).
• If the contents of the accordion box are obvious, then it's OK to hide the title (for either aesthetic or vertical space reasons), since the purpose of the title is to remind the user what's in the collapsed box.  This does not introduce any additional cognitive load for the user.

**Comment by terracoda** [comment]
@pixelzoom, I think we are all in agreement.
